File: lib/Models/GetModelsResponse.cpp


#include "GetModelsResponse.h"

using namespace Tiny;

GetModelsResponse::GetModelsResponse()
{
	models = std::list<ModelDTO>();
	currency = CurrencyType();
	regionId = long(0);
}

GetModelsResponse::GetModelsResponse(std::string jsonString)
{
	this->fromJson(jsonString);
}

GetModelsResponse::~GetModelsResponse()
{

}

void
GetModelsResponse::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *modelsKey = "models";

    if(object.has_key(modelsKey))
    {
        bourne::json value = object[modelsKey];


        std::list<ModelDTO> models_list;
        ModelDTO element;
        for(auto& var : value.array_range())
        {


            element.fromJson(var.dump());

            models_list.push_back(element);
        }
        models = models_list;


    }

    const char *currencyKey = "currency";

    if(object.has_key(currencyKey))
    {
        bourne::json value = object[currencyKey];




        CurrencyType* obj = &currency;
		obj->fromJson(value.dump());

    }

    const char *regionIdKey = "regionId";

    if(object.has_key(regionIdKey))
    {
        bourne::json value = object[regionIdKey];



        jsonToValue(&regionId, value, "long");


    }


}

bourne::json
GetModelsResponse::toJson()
{
    bourne::json object = bourne::json::object();




    std::list<ModelDTO> models_list = getModels();
    bourne::json models_arr = bourne::json::array();

    for(auto& var : models_list)
    {
        ModelDTO obj = var;
        models_arr.append(obj.toJson());
    }
    object["models"] = models_arr;








	object["currency"] = getCurrency().toJson();





    object["regionId"] = getRegionId();



    return object;

}

std::list<ModelDTO>
GetModelsResponse::getModels()
{
	return models;
}

void
GetModelsResponse::setModels(std::list <ModelDTO> models)
{
	this->models = models;
}

CurrencyType
GetModelsResponse::getCurrency()
{
	return currency;
}

void
GetModelsResponse::setCurrency(CurrencyType  currency)
{
	this->currency = currency;
}

long
GetModelsResponse::getRegionId()
{
	return regionId;
}

void
GetModelsResponse::setRegionId(long  regionId)
{
	this->regionId = regionId;
}

File: src/main/java/org/openapitools/server/model/GetModelsResponse.java
package org.openapitools.server.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import org.openapitools.server.model.CurrencyType;
import org.openapitools.server.model.ModelDTO;
import jakarta.validation.constraints.*;
import jakarta.validation.Valid;


/**
 * Ответ на запрос информации о моделях.
 */
public class GetModelsResponse   {

    private List<@Valid ModelDTO> models = new ArrayList<>();
    private CurrencyType currency;
    private Long regionId;

    /**
     * Default constructor.
     */
    public GetModelsResponse() {
    // JSON-B / Jackson
    }

    /**
     * Create GetModelsResponse.
     *
     * @param models Список моделей товаров.
     * @param currency currency
     * @param regionId Идентификатор региона, для которого выводится информация о предложениях модели (доставляемых в этот регион).  Информацию о регионе по идентификатору можно получить с помощью запроса [GET regions/{regionId}](../../reference/regions/searchRegionsById.md). 
     */
    public GetModelsResponse(
        List<@Valid ModelDTO> models, 
        CurrencyType currency, 
        Long regionId
    ) {
        this.models = models;
        this.currency = currency;
        this.regionId = regionId;
    }



    /**
     * Список моделей товаров.
     * @return models
     */
    public List<@Valid ModelDTO> getModels() {
        return models;
    }

    public void setModels(List<@Valid ModelDTO> models) {
        this.models = models;
    }

    /**
     * Get currency
     * @return currency
     */
    public CurrencyType getCurrency() {
        return currency;
    }

    public void setCurrency(CurrencyType currency) {
        this.currency = currency;
    }

    /**
     * Идентификатор региона, для которого выводится информация о предложениях модели (доставляемых в этот регион).  Информацию о регионе по идентификатору можно получить с помощью запроса [GET regions/{regionId}](../../reference/regions/searchRegionsById.md). 
     * @return regionId
     */
    public Long getRegionId() {
        return regionId;
    }

    public void setRegionId(Long regionId) {
        this.regionId = regionId;
    }

    /**
      * Create a string representation of this pojo.
    **/
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class GetModelsResponse {\n");
        
        sb.append("    models: ").append(toIndentedString(models)).append("\n");
        sb.append("    currency: ").append(toIndentedString(currency)).append("\n");
        sb.append("    regionId: ").append(toIndentedString(regionId)).append("\n");
        sb.append("}");
        return sb.toString();
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
    */
    private static String toIndentedString(Object o) {
        if (o == null) {
          return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }
}

